SwissDataScienceCenter/renku-ui
# feat(client): improve starting new sessions

This PR fixes a number of problems with starting sessions, and it overhauls the logic to verify docker image availability.

**CI:**
Now we always check for the final image to be available in every case.
If the user is logged, we first verify that pipelines and jobs have run as expected.
If the user has permissions, we show button to restart jobs and try to fix the most common project on the fly.

**FIXES:**
Visual glitches and inconsistent results should be fixed. Some of the most visible are:
- the fields "Branch" and "Commit" are never shown empty.
- changing branch wait for all the data to be available before flashing a commit and changing it
- autostarting a notebook has a consistent behavior, and show the error when one happens


**How to test**
I created a number of projects (name starting with "1408-") to test the different cases, but covering all the cases may be tricky.
Try to mess up projects or removing previously created images is an effective way of testing.
Also, verifying if a case is missing from the tests would be good.

Here are 2 fairly common cases that used to create problem -- i.e. images marked as available but actually not available:
- Job finished, but image not available (logged user): https://dev.renku.ch/projects/lorenzo.cavazzi.tech/1408-missing-image/sessions/new
- Pinned image not available (all uswers): https://dev.renku.ch/projects/lorenzo.cavazzi.tech/custom-image/sessions/new

/deploy renku=000-acceptance-tests #persist
fix #1408
fix #1679
